src/lightning_app/components/auto_scaler.py

```diff
@@ -0,0 +1,529 @@
+            "accept": "application/json",
+            "username": USERNAME,
+            "Authorization": AUTHORIZATION_TYPE + " " + data,
+        }
+        response = requests.put(f"{self.url}/system/update-servers", json=servers, headers=headers, timeout=10)
+        response.raise_for_status()
+
+
+class AutoScaler(LightningFlow):
+    """A LightningFlow component that handles all the servers and uses load balancer to spawn up and shutdown based
+    on current requests in the queue.
+
+    Args:
+        min_replicas: Number of works to start when app initializes.
+        max_replicas: Max numbers of works to spawn to handle the incoming requests.
+        autoscale_interval: Number of seconds to wait before checking whether to upscale or downscale the works.
+        max_batch_size: Number of requests to process at once.
+        timeout_batch: Number of seconds to wait before sending the requests to process.
+        downscale_threshold: Lower limit to determine when to stop works.
+        upscale_threshold: Upper limit to determine when to spawn up a new work.
+        worker_url: Default=api/predict. Provide the REST API path
+        input_schema:
+        output_schema:
+
+        .. doctest::
+
+            >>> from lightning_app.components import AutoScaler
+            >>> from lightning_app import LightningApp
+            >>> app = LightningApp(
+            ...     AutoScaler(
+            ...         MyPythonServer,  # noqa: F821
+            ...         min_replicas=1,
+            ...         max_replicas=4,
+            ...         autoscale_interval=10,
+            ...     )
+            ... )
+    """
+
+    def __init__(
+        self,
+        work_cls: Optional[type] = None,
+        min_replicas: int = 1,
+        max_replicas: int = 4,
+        autoscale_interval: int = 1 * 10,
+        max_batch_size: int = 8,
+        timeout_batch: float = 2,
+        downscale_threshold: Optional[int] = None,
+        upscale_threshold: Optional[int] = None,
+        worker_url: str = None,
+        input_schema: Any = Dict,
+        output_schema: Any = Dict,
+    ) -> None:
+        super().__init__()
+        self.num_replicas = 0
+        self._work_registry = {}
+
+        assert work_cls is not None or is_overridden("create_worker", self, AutoScaler)
+        self._work_cls = work_cls
+        self._input_schema = input_schema
+        self._output_schema = output_schema
+        self.autoscale_interval = autoscale_interval
+
+        if max_replicas < min_replicas:
+            raise ValueError("max_replicas must be less than or equal to min_replicas.")
+        self.max_replicas = max_replicas
+        self.min_replicas = min_replicas
+        self.downscale_threshold = downscale_threshold or min_replicas
+        self.upscale_threshold = upscale_threshold or min_replicas * max_batch_size
+        self._last_autoscale = time.time()
+        self.fake_trigger = 0
+
+        worker_url = worker_url or "api/predict"
```

Comment:
```suggestion
        worker_path = worker_url or "api/predict"
```

Reply:
Let's continuing this in https://github.com/Lightning-AI/lightning/pull/15769#discussion_r1034730713.